codebase-manager/plugins/plugin.php

```diff
@@ -43,6 +47,46 @@ public function display_version_update_information( int $number_of_columns ): vo
 		print( '</p></div></td></tr>' );
 	}
 
+	public function display_vulnerability_information( int $number_of_columns ): void {
+		if ( ! $this->has_vulnerabilities() ) {
+			return;
+		}
+
+		printf( '<tr class="plugin-vuln-tr%s">', $this->is_active() ? ' active' : '' );
+		printf( '<td colspan="%s" class="plugin-vuln colspanchange">', esc_attr( $number_of_columns ) );
+		print( '<div class="vuln-message error inline notice-error notice-alt">' );
+
+		/* translators: 1: The number of vulnerabilities associated with the plugin. */
+		$message = sprintf( _n(
+			'There is %s known vulnerability for this plugin on the currently installed version:',
+			'There are %s known vulnerabilities for this plugin on the currently installed version:',
+			count( $this->vulnerabilities )
+		), number_format_i18n( count( $this->vulnerabilities ) ) );
+
+		echo '<p>' . esc_html( $message ) . '</p>';
+
+		echo '<ul>';
+		foreach ( $this->vulnerabilities as $vuln ) {
+			// "None" could be confusing, let's convert it to "Low".
+			$severity = 'NONE' === $vuln->severity ? 'Low' : ucwords( strtolower( $vuln->severity ) );
+
+			$severity_info = sprintf( '%s severity', esc_html( $severity ) );
+			if ( null !== $vuln->severityScore ) {
+				$severity_info = sprintf( '%s severity: %s/10', esc_html( $severity ), esc_html( $vuln->severityScore ) );
+			}
```

Comment:
If we use `_n()` for `$message`, it probably makes sense to use translation functions here as well.